new-dti-website/components/team/MemberGroup.tsx

```diff
@@ -0,0 +1,326 @@
+                          alt={icon.alt}
+                          height={icon.height}
+                          width={icon.width}
+                        />
+                      </a>
+                    )
+                  );
+                })}
+              </div>
+            </div>
+            <div className="md:block xs:hidden">
+              <button
+                onMouseEnter={mouseHandler}
+                onMouseLeave={mouseHandler}
+                className="py-3 px-5 bg-white rounded-xl text-[#A52424] border-[3px] border-[#A52424] 
+              hover:bg-[#A52424] hover:text-white stroke-white"
+              >
+                <div className="flex gap-3 w-max">
+                  <Image
+                    src="/icons/calendar.svg"
+                    alt="calendar"
+                    width={24}
+                    height={24}
+                    className={hover ? 'brightness-0 invert' : ''}
+                  />
+                  <p className="font-bold text-lg text-inherit whitespace-nowrap">Chat with me</p>
+                </div>
+              </button>
+            </div>
+          </div>
+        </div>
+        <div onClick={props.onClose}>
+          <Image
+            src="/icons/close.svg"
+            width={23}
+            height={23}
+            alt="close"
+            className="m-2 cursor-pointer xs:w-4"
+          />
+        </div>
+      </div>
+      <div className="md:hidden xs:block">
+        <button
+          onMouseEnter={mouseHandler}
+          onMouseLeave={mouseHandler}
+          className="py-3 px-5 bg-white rounded-xl text-[#A52424] border-[3px] border-[#A52424] 
+              hover:bg-[#A52424] hover:text-white stroke-white w-full flex justify-center"
+        >
+          <div className="flex gap-3 w-max">
+            <Image
+              src="/icons/calendar.svg"
+              alt="calendar"
+              width={24}
+              height={24}
+              className={hover ? 'brightness-0 invert' : ''}
+            />
+            <p className="font-bold text-base text-inherit whitespace-nowrap">Chat with me</p>
+          </div>
+        </button>
+      </div>
+    </Card>
+  );
+};
+
+type MemberGroupProps = {
+  roleName: string;
+  description: string;
+  members: IdolMember[];
+  setSelectedMember: Dispatch<SetStateAction<IdolMember | undefined>>;
+  selectedMember: IdolMember | undefined;
+  selectedRole: string;
+  memberDetailsRef: RefObject<HTMLInputElement>;
+};
+
+const MemberGroup: React.FC<MemberGroupProps> = ({
+  roleName,
+  description,
+  members,
+  setSelectedMember,
+  selectedMember,
+  selectedRole,
+  memberDetailsRef
+}) => {
+  const selectedMemberIndex: number = useMemo(
+    () => (selectedMember ? members.indexOf(selectedMember) : -1),
+    [members, selectedMember]
+  );
+
+  const { width } = useScreenSize();
+
+  const canInsertMemberDetails = (index: number): boolean => {
+    let columns = 4;
```

Comment:
Can you leave a comment explaining what `columns` and `index` is and what's going on here in general as well (especially with the checks we're doing underneath)?

Also we shouldn't really be using magic numbers `768` and `1024`, and `2`, `3`, `4`. Instead, let's define consts so that these numbers are self-documenting (i.e. I know what each number means).

Nit: it might be make sense to use a switch/case here instead of if/else.